### ISSUE TITLE
Remove unions in MediaTrackCapabilities

### ISSUE THREAD

**Issue by jan-ivar** [w3c/mediacapture-main#408]
Unions may be convenient in input, but add unnecessary variance tripping up code on output.

A parsing nightmare, even with destructuring:

``` js
let {width, height} = track.getCapabilities();
let maxWidth = (typeof width == "object")? width.max : width;
let maxHeight = height.max || height; // works in some cases, not booleans though
```

We should remove all unions in [MediaTrackCapabilities](https://w3c.github.io/mediacapture-main/getusermedia.html#idl-def-mediatrackcapabilities) like this:

``` js
dictionary MediaTrackCapabilities {
    LongRange           width;
    LongRange           height;
    DoubleRange         aspectRatio;
    DoubleRange         frameRate;
    sequence<DOMString> facingMode;
    DoubleRange         volume;
    LongRange           sampleRate;
    LongRange           sampleSize;
    sequence<boolean>   echoCancellation;
    DoubleRange         latency;
    LongRange           channelCount;
    DOMString           deviceId;
    DOMString           groupId;
};
```

For straight up parsing:

``` js
let {width: {max: maxWidth}, height: {max: maxHeight}} = track.getCapabilities();
```

**Comment by jan-ivar** [comment]
> ```js
let maxHeight = height.max || height; // works in some cases, not booleans though
```

And that doesn't even work. Need tricks like `let maxHeight = height.max || height > 0 && height`

**Comment by alvestrand** [comment]
This seems like a good idea to me. Can you make a) a PR, and b) a note to the list about it?

(Chrome hasn't implemented capabilities yet)

**Comment by jan-ivar** [comment]
Done.

**Comment by alvestrand** [comment]
Resolved by #415
